### DISCUSSION TITLE
How the automation/order works

### DISCUSSION THREAD

**Discussion by chuccck** [seanap/Plex-Audiobook-Guide#47] [Q&A]
I am bit lost on the automation workflow of this.  Should I be converting the mp3s to m4b first, then have mp3tag update that one m4b file?  Or vice a versa?

how do you set mp3tag to automate the process?

I used the program below to convert to m4b as it seemed much simpler, but I dont think there is a way to automate it.  Do you think its possible?

edit:
forgot program:  https://github.com/yermak/AudioBookConverter

2nd edit: is there a Windows script available for the m4b tool?  I'm windows only

**Comment by rabelux** [comment]
Which program below do you mean? Or is the program's name "below"

**Reply by chuccck** [reply]
> Which program below do you mean? Or is the program's name "below"

edited my op but this one: https://github.com/yermak/AudioBookConverter

**Comment by seanap** [comment]
Here is an overview of how my process flows:

1. Acquire a new book and save to ../original
2. If m4b then dropit copies the book to ../temp
3. If mp3 then dropit copies it to ../mp3ab and the auto-m4b-tool script watches this folder and converts to m4b and moves the file to ../temp
4. Open mp3tag which loads everything in ../temp
5. Run the audible metadata scraper (`ctrl+shift+i`)
6. Run the mp3tag action script which moves everything to the right folder and exports all helper files (`alt+a 1`)

A lot of mp3 to m4b converters strip out some metadata/covers so I always convert first then run it though the tagger.

Mp3tag cannot be fully automated.  You need to run steps 5 & 6 for each book.  The keyboard shortcuts make it quick and easy, but you may have books that aren't in Audible or Audible's data may be slightly wrong and you need to adjust.  Since there isn't an audiobook database (like imdb/tmdb/tvdb/etc.) and the original source files may not be named correctly or may have incorrect or missing tags it is very hard to fully automate tagging.  It all depends on your source material.  Are you ripping directly from your audible? Then you probably have the correct ALBUM and ARTIST tag's and you don't need to use mp3tag at all.

You can use what ever converter program you want.  I chose to use m4b-tool because I can automate it with scripts.  If you are running windows 10 or 11 then you can run Linux Dockers.  If you would like to set up the same automation I have check out this docker someone made from my auto-m4b-tool script https://hub.docker.com/r/9madmax5/m4b-tool

Here are the major take aways - Plex needs the ALBUM and ARTIST tag correct to import the book.  If you have these tags then there isn't much you need to do, it should just work.  By using mp3tag we set the tags to exactly what the Plex agent is looking for which almost guarantee's a correct match.  By converting to m4b's your total file count of your library is kept very low and everything runs smoother if you have a large library (small libraries it doesn't really matter much).

**Reply by chuccck** [reply]
thanks for the detailed writeup.  I am on windows 10 and would be brand new to docker in general, but see they have it for windows.  I am interested in the automation but would my to find a guide to walk me through it in docker.

it looks like your process is still manual during mp3tag right?  And you can't run more than 1 book through mp3tag at a time?  

But you can batch convert multiple books with the m4btool assuming I get it running in docker...?

What do you do if you get a book that has chapterized m4b files or multiple m4b files?

**Reply by seanap** [reply]
Before I wrote the audible web scraping script "manual" meant literally typing all the information into each tag and manually creating the folder structures and moving the files.  Having a script that scrapes audible for the tags and another script that moves the files based on those tags is at the very least a step in the right direction.  I'm all ears if you have a better way.  Like I said it all depends on your source material, if your tags and folders are good then no need for mp3tag.  

In order to fully automate the tagger needs to know exactly which book and edition it's editing, this means either the folder or file names need to be able to be parsed, conflicts need to be resolved (books with multiple narrators or abridged versions etc.), and it needs a database to pull the information from that goes beyond just Audible as a source.  Currently nothing exists that can do this. 

Yes, m4b-tool can batch convert as long as each book is in it's own folder.

I turn the mp3's into chapterized m4b's.  This is the preferred format.  Single file m4b's just get copied to the folder mp3tag monitors. M4b-tool can also combine multi-m4b files, but those are more rare.  When those come up I just toss them into the folder auto-m4b-tool looks at and it combines them.

Nothing about my process or guide is the end all be all perfect solution.  It's just a combination of the best available tools to get everything in the right format with the least amount of effort.  As new programs are developed I will update the guide.

**Reply by chuccck** [reply]
of course, let me apoligize if I am coming off ungrateful.  This guide and your work is truly appreciated, I just wanted to understand which actions I need to take and which are done without my input.  

OK so if I am all windows, you think that docker container you linked is the best bet?  I literally know nothing about Docker or Docker for windows but am willing to try, just don't know what to do first.  The audiobook converter app I downloaded is pretty limited honestly and would like to figure out your script.

**Reply by seanap** [reply]
No your weren't coming off like that at all! I've been asking everyone and compiling different methodologies for many years, and I really can't wait for the day when the guide in it's current form is obsolete. We are so close to having a truly automated set up, a lot of the individual pieces exist but nothing that combines them all.

I've never used WSL or linux dockers on Windows so I don't have a guide that's worked for me, I have just heard of other people who have had success going that route. It depends on volume if it will be worth it. 100 books? Audiobookconverter will probably be just fine. 10,000 books? You'll want an automated batch converter.

**Reply by chuccck** [reply]
i definitely have less than 100 books right now, probably makes sense to muddle through using audiobook converter
